### ISSUE TITLE
Allow challenger to provide a corrected version of the translation

### ISSUE THREAD

**Issue by hbarcelos** [kleros/linguo#22]
As suggested by @fnanni-0 [here](https://github.com/kleros/linguo-contracts/pull/10#issuecomment-731424762):
> I think it would be great if the challenger could submit a translation with the corrections and claim the task price too. If the challenger wins, not only the translator is punished and the challenger is rewarded, but the requester gets the translation he requested instead of losing time and money.

This could improve requester's UX and potentially increase the payout of challengers.

Currently challengers can profit by finding a single issue with the translation which would make it non-compliant with the required quality tier. One can optimize this task by scanning for common sources of mistakes, for example.

With this change, if the challenger decides to put some additional effort into the task to provide a fully compliant translation, beyond getting the requester deposit, he would also be rewarded with the escrow payment from the requester.

The requester benefits because she would get the translation done. Currently she gets refunded of her initial escrow deposit, but will spend both money with gas and time waiting for the task. If she has a deadline, this might be a deal-breaker.

Some challenges remain on how to actually implement this:

- Should we extend the current flow to allow the challenger to become the translator?
    - How much complexity would that bring to the smart contract?
    - Should we allow this process to be recurrent (a 4th party to challenge the original challenger's corrected translation)?
        - How to prevent and endless cycle of challenging that would keep the requester's escrow deposit locked in the contract for a long period of time?
- Should we change the dispute ruling options to be non-binary and give jurors the ability to choose to accept the challenger's translation instead of the original translator's? (Suggestion made by @fnanni-0)

**Comment by epiqueras** [comment]
>Should we change the dispute ruling options to be non-binary and give jurors the ability to choose to accept the challenger's translation instead of the original translator's? (Suggestion made by @fnanni-0)

This makes the most sense to me. Otherwise, the process could become even slower.

**Comment by mtsalenc** [comment]
I really like this. One strategy challengers use in the t2cr is to only challenge regarding one mistake so that they can profit again later. I think this would discourage this behavior.

I believe it would change quite a bit of the contract, however I think we'd need add some recursion to allow challenges to the challenger submission and so on. 

I'd say spend some time on the contract and see if it gets too complicated.

**Comment by fnanni-0** [comment]
Another thing that could be interesting, but has the drawback of adding even more complexity to the contracts, is to allow some kind of settlement outside of court. 

For example, if I submitted a translation and someone challenged it with an improved version of it, I could admit my mistake and share the task price with the challenger instead of going through the arbitration process with the certainty that I'm going to lose. One problem of this though, is that I would have incentives to challenge my own submission unless we allow the challenger to also get challenged. And maybe incentives for sloppy submissions would increase too.

**Comment by fnanni-0** [comment]
I think of challengers as first instance judges (or something like that)

**Comment by epiqueras** [comment]
That's like a snake that eats its tail.

The non-binary ruling still makes the most sense.

**Comment by hbarcelos** [comment]
Yeah, I think having 3 ruling options would suffice:

1. Approve the original translation.
2. Reject the original translation.
3. Approve the translation provided by the challenger.

Changes in contract would be significant, but not in a sense that everything would have to be done from scratch.

**Comment by fnanni-0** [comment]
Now that I think about it 2. and 3. are a subset of "reject the original translation" and that could be a problem. How does an arbitrator deal with the following situation: 

- 4 jurors vote 1.
- 3 jurors vote 2.
- 2 jurors vote 3.

Here 1. wins, but there are 5 jurors voting for options in which the original translation gets rejected.

**Comment by epiqueras** [comment]
A policy can say to pick the provided translation, if suitable.

**Comment by hbarcelos** [comment]
> Here 1. wins, but there are 5 jurors voting for options in which the original translation gets rejected.

Well, that's how a voting with 3+ options work. If it is not required a simple majority to determine the winner, the most voted option wins.

Depending on the way the policy is phrased, those 3 options can be considered mutually-exclusive. I don't think that's an issue, it's how it should work.

**Comment by clesaege** [comment]
Yeah, it could be an option and he would have the evidence period to make the new translation.
